x-pack/plugins/observability_solution/infra/public/components/asset_details/add_metrics_callout/constants.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ObservabilityOnboardingLocatorParams } from '@kbn/deeplinks-observability';
+import { i18n } from '@kbn/i18n';
+import { AddDataPanelProps } from '@kbn/observability-shared-plugin/public';
+import { LocatorPublic } from '@kbn/share-plugin/common';
+import { OnboardingFlow } from '../../shared/templates/no_data_config';
+
+export type AddMetricsCalloutKey =
+  | 'hostOverview'
+  | 'hostMetrics'
+  | 'hostProcesses'
+  | 'containerOverview'
+  | 'containerMetrics';
+
+const defaultPrimaryActionLabel = i18n.translate(
+  'xpack.infra.addDataCallout.hostOverviewPrimaryActionLabel',
+  {
+    defaultMessage: 'Add Metrics',
+  }
+);
+
+const defaultContent = {
+  content: {
+    title: i18n.translate('xpack.infra.addDataCallout.defaultTitle', {
+      defaultMessage: 'View core metrics to understand your host performance',
+    }),
+    content: i18n.translate('xpack.infra.addDataCallout.defaultContent', {
+      defaultMessage:
+        'Collect metrics such as CPU and memory usage to identify performance bottlenecks that could be affecting your users.',
+    }),
+  },
+};
+
+const hostDefaultActions = (
+  locator: LocatorPublic<ObservabilityOnboardingLocatorParams> | undefined
+) => {
+  return {
+    actions: {
+      primary: {
+        href: locator?.getRedirectUrl({ category: OnboardingFlow.Hosts }) || '',
```

Comment:
Why are you fall backing to and **empty string** in case the locator is undefined? Doesn't it make sense to make `href` undefined as well?

Reply:
After a brief private discussion, we decided to allow the `href` to be `string | undefined`, as we shouldn’t display an action button if no href is defined in order to avoid delivering a poor UX experience to users.

I’d like to add that the scenario where the locator isn’t defined shouldn’t be common. It would only happen if someone modified the locators and, for some reason, there was no match with the ID.